source/common/network/dns_impl.cc

```diff
@@ -168,15 +168,16 @@ void DnsResolverImpl::PendingResolution::onAresGetAddrInfoCallback(int status, i
 
   if (completed_) {
     if (!cancelled_) {
-      try {
-        callback_(resolution_status, std::move(address_list));
-      } catch (const EnvoyException& e) {
+      TRY_NEEDS_AUDIT { callback_(resolution_status, std::move(address_list)); }
```

Comment:
This should be main thread only.

Reply:
There is a filter called dns_filter that use this...

Reply:
Pretty positive this is main thread only. What is the callsite you are concerned about?

Reply:
In IpVersions/DnsFilterIntegrationTest.ExternalLookupTest/IPv4, it is called on worker thread.

Reply:
Let's capture this in a comment or TODO and see if we can in the future try to eliminate this exception-path also.